s3path.py

```diff
@@ -576,6 +581,23 @@ def replace(self, target):
         """
         return self.rename(target)
 
+    def unlink(self, missing_ok=False):
+        """
+        Remove this key from its bucket.
+        """
+        self._absolute_path_validation()
+        # S3 doesn't care if you remove full prefixes or buckets with its delete API
+        # so unless we manually check, this call will be dropped through without any
+        # validation and could result in data loss
+        if self.is_dir():
+            raise IsADirectoryError(str(self))
+        # XXX: Note: If we don't check if the file exists here, S3 will always return
+        # success even if we try to delete a key that doesn't exist. So, if we want
```

Comment:
Like in the comment, I noted that S3 doesnt care if the file exists and will always return success, so I reasoned that it doesn't matter if the file doesn't exist and wouldn't be helpful to raise an error in that case.

If you want to match the exact pathlib API no matter what (and use an extra API call), I can add this check, but it will slow down the usage of this library.

Reply:
I see your point.